papermill_origami/engine.py

```diff
@@ -288,6 +295,67 @@ async def async_execute_cell(
         #     or "raises-exception" in cell.metadata.get("tags", []))
 
         # By default this will wait until the cell execution status is no longer active
+
+        async def update_outputs_callback(resp: UpdateOutputCollectionEventSchema):
+            """Callback to set cell outputs observed from Noteable over RTU into the
+            corresponding cell outputs here in Papermill
+
+            TODO: This callback currently only sets error outputs. We will need to extend it to set all types of outputs
+                  with a translation layer from Noteable to Jupyter.
+            """
+            if resp.data.cell_id != cell.id:
+                raise SkipCallback("Not tracked cell")
+            if not resp.data.outputs:
+                raise SkipCallback("Nothing to do")
+
+            for output in resp.data.outputs:
+                # We need to map parent_collection_id to cell_id in order to process any append_output_events
+                # which are uniquely identified by collection_id and not cell_id
+                self.__noteable_output_collection_cache[output.parent_collection_id] = cell.id
+                if output.type == KernelOutputType.error:
+                    if output.content.raw:
+                        error_data = orjson.loads(output.content.raw)
+                        error_output = nbformat.v4.new_output("error", **error_data)
+                        self.nb.cells[cell_index].outputs.append(error_output)
+                        return True
```

Comment:
We'll probably want to change the return pattern when the TODO above is implemented